ext/stringio/stringio.c

```diff
@@ -1859,7 +1860,12 @@ strio_set_encoding(int argc, VALUE *argv, VALUE self)
 	}
     }
     ptr->enc = enc;
-    if (!NIL_P(ptr->string) && WRITABLE(self)) {
+    if (!NIL_P(ptr->string) && WRITABLE(self)
+#if RUBY_API_VERSION_MAJOR == 3 || RUBY_API_VERSION_MAJOR >= 4
```

Comment:
```suggestion
#if (RUBY_API_VERSION_MAJOR == 3 && RUBY_API_VERSION_MINOR >= 4) || RUBY_API_VERSION_MAJOR >= 4
```

? (If we want to use this only for Ruby 3.4+.)

Reply:
Thank you for catching this.  I think this is fine.  My original idea (other than the typo), was :

```c
#if RUBY_API_VERSION_MAJOR == 3 && RUBY_API_VERSION_MINOR >= 4
```

with the expectation that by Ruby 4, we will no longer have chilled strings, and literal strings will be frozen by default.

Reply:
Ah, I see.

(I'll hear from Matz about literal strings at RubyKaigi 2025...)